### PR TITLE
Add ResourceHelper

### PR DESCRIPTION
**Pull request by gjojgov-evolution**

### CODE REVIEW

File: core/src/main/scala/com/evolutiongaming/catshelper/ResourceHelper.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package com.evolutiongaming.catshelper
+
+import cats.implicits._
+import cats.effect.{Resource, Sync}
+
+object ResourceHelper {
```
Comment (t3hnar) [inline core/src/main/scala/com/evolutiongaming/catshelper/ResourceHelper.scala:6]: why not put this under `CatsHelper` ?

File: core/src/main/scala/com/evolutiongaming/catshelper/ResourceHelper.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        timedRelease = for {
+          getMeasurement <- MeasureDuration[F].start
+          _              <- Log[F].info(s"$name releasing")
+          _ <- release.attemptTap {
+            case Left(err) => for {
+              measureResult <- getMeasurement
+              _ <- Log[F].error(s"$name release failed in ${measureResult.toMillis}ms with $err", err)
+            } yield ()
+
+            case Right(_) => for {
```
Comment (t3hnar) [inline core/src/main/scala/com/evolutiongaming/catshelper/ResourceHelper.scala:37]: we un-match `Right(a)` and use `a` in `yield` :)

**Review by t3hnar — CHANGES_REQUESTED** [review]

File: core/src/main/scala/com/evolutiongaming/catshelper/ResourceHelper.scala (hunk 2 above)
Comment (gjojgov-evolution) [reply core/src/main/scala/com/evolutiongaming/catshelper/ResourceHelper.scala]: Not sure I understand the comment correctly, but have a look at the updated version.

File: core/src/main/scala/com/evolutiongaming/catshelper/ResourceHelper.scala (hunk 1 above)
Comment (gjojgov-evolution) [reply core/src/main/scala/com/evolutiongaming/catshelper/ResourceHelper.scala]: Yep, moved!

**Review by t3hnar — APPROVED** [review]

**Pull request merged**